modules/navier_stokes/src/fvbcs/PINSFVMomentumAdvectionOutflowBC.C

```diff
@@ -11,82 +11,39 @@
 #include "PINSFVSuperficialVelocityVariable.h"
 #include "SubProblem.h"
 #include "MooseMesh.h"
+#include "NS.h"
 
-registerADMooseObject("NavierStokesApp", PINSFVMomentumAdvectionOutflowBC);
+registerMooseObject("NavierStokesApp", PINSFVMomentumAdvectionOutflowBC);
 
 InputParameters
 PINSFVMomentumAdvectionOutflowBC::validParams()
 {
-  InputParameters params = FVMatAdvectionOutflowBC::validParams();
-  params += INSFVFullyDevelopedFlowBC::validParams();
-  params.addRequiredCoupledVar("u", "The velocity in the x direction.");
-  params.addCoupledVar("v", "The velocity in the y direction.");
-  params.addCoupledVar("w", "The velocity in the z direction.");
-  params.addRequiredCoupledVar("porosity", "Porosity auxiliary variable");
+  InputParameters params = INSFVMomentumAdvectionOutflowBC::validParams();
+  params.addRequiredParam<MooseFunctorName>(NS::porosity, "Porosity auxiliary variable");
   params.addClassDescription("Outflow boundary condition for advecting momentum. This will impose "
                              "a zero normal gradient on the boundary velocity.");
```

Comment:
```suggestion
  params.addClassDescription("Outflow boundary condition for advecting momentum in the porous media momentum equation "
                                                     "This will impose a zero normal gradient on the boundary velocity.");
```